### PR TITLE
Remove redundant path validation

### PR DESCRIPTION
**Pull request by rht**
Because path.ParsePath is always called whenever Resolve is called

### CODE REVIEW

**Comment by whyrusleeping** [comment]
I'm not seeing where the path validation is happening in `Resolve`. (I'm also pre-coffee-reading through this, so i may have missed something blatant)

**Comment by rht** [comment]
https://github.com/ipfs/go-ipfs/blob/master/core/pathresolver.go#L38 -> https://github.com/ipfs/go-ipfs/blob/master/path/path.go#L47-L49. ParsePath is reused a lot.

File: core/commands/get.go [hunk 1]
```diff
@@ -65,14 +65,8 @@ may also specify the level of compression by specifying '-l=<1-9>'.
 			return
 		}
 
-		// Validate path string
-		p, err := path.ParsePath(req.Arguments()[0])
-		if err != nil {
-			res.SetError(fmt.Errorf("failed to validate path: %v", err), cmds.ErrNormal)
-			return
-		}
+		p := path.Path(req.Arguments()[0])
```
Comment (jbenet) [inline core/commands/get.go]: here we want the validation, because it gives the user a good error (path invalid)

File: core/commands/publish.go [hunk 2]
```diff
@@ -88,15 +88,9 @@ Publish an <ipfs-path> to another public key (not implemented):
 			pstr = args[0]
 		}
 
-		p, err := path.ParsePath(pstr)
-		if err != nil {
-			res.SetError(fmt.Errorf("failed to validate path: %v", err), cmds.ErrNormal)
-			return
-		}
-
```
Comment (jbenet) [inline core/commands/publish.go]: same here-- validating beforehand gives the user a good error.

**Comment by jbenet** [comment]
@rht i see that `Resolve` will ParsePath and return a path validation error. Can we maybe test these commands to make sure they work that way? it may be ok to make these changes

**Comment by rht** [comment]
Looks like `Resolve` calls `ParsePath` only when the prefix is ipns. Will change the code accordingly.

File: path/resolver.go [hunk 3]
```diff
@@ -65,6 +65,11 @@ func SplitAbsPath(fpath Path) (mh.Multihash, []string, error) {
 // ResolvePath fetches the node for given path. It returns the last item
 // returned by ResolvePathComponents.
 func (s *Resolver) ResolvePath(ctx context.Context, fpath Path) (*merkledag.Node, error) {
+	// validate path
+	if err := fpath.IsValid(); err != nil {
```
Comment (rht) [inline path/resolver.go]: Validation is run here instead of outside of any call to `Resolver.ResolvePath` / `Core.Resolve`.

Comment (jbenet) [inline path/resolver.go]: ok sounds good

**Comment by jbenet** [comment]
tests failed, including

```
--- FAIL: TestResolveNoComponents (0.03s)
	pathresolver_test.go:24: Should error with no components (/ipfs/). not a key
FAIL
FAIL	github.com/ipfs/go-ipfs/core	0.042s

```

**Comment by rht** [comment]
Yes saw that. Fixed. But perhaps `ErrNoComponents` should appear only once in the code. The travis err is the occasional error with mock_notif_test.go, else passes.

**Comment by jbenet** [comment]
thanks @rht

**Pull request merged**